src/gp_engine.hpp

```diff
@@ -79,6 +79,9 @@ class GPEngine {
   // Use branch lengths from loaded sample as a starting point for optimization.
   void HotStartBranchLengths(const RootedTreeCollection& tree_collection,
                              const BitsetSizeMap& indexer);
+  // Gather branch lengths from loaded sample with their corresponding pcsp.
+  SizeDoubleVector GatherBranchLengths(const RootedTreeCollection& tree_collection,
```

Comment:
Rather than returning a SizeDoubleVector of all of the branch lengths jumbled up, it would seem nicer to return a map from size (PCSP) to a double vector, collating all of the branch lengths together for a given PCSP. Is there a reason we shouldn't do that?